hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/ActionRepository.java

```diff
@@ -207,6 +207,16 @@
     void switchStatus(@Param("statusToSet") Action.Status statusToSet, @Param("targetsIds") List<Long> targetIds,
             @Param("active") boolean active, @Param("currentStatus") Action.Status currentStatus);
 
+    @Query("SELECT a.id FROM JpaAction a WHERE a.target IN :targetsIds AND a.active = :active AND a.status = :currentStatus AND a.distributionSet.requiredMigrationStep = false")
```

Comment:
check if entity graph os needed (adapt other queries correspondingly)

Reply:
In this case Entity Graph is not needed, because we are interested only in Action IDs. On the other hand, other queries return multiple JpaAction entities and use Entity Graph in order to eagerly load corresponding targets / distribution sets resolving the N + 1 problem, when referencing them in a loop / stream.